content/en/docs/components/pipelines/sdk/component-development.md

```diff
@@ -1,85 +1,93 @@
 weight = 40
-
 +++
-{{% alert title="Out of date" color="warning" %}}
-This guide contains outdated information pertaining to Kubeflow 1.0. This guide
-needs to be updated for Kubeflow 1.1.
-{{% /alert %}}
-
-This page describes how to author a reusable component that you can
-load and use as part of a pipeline.
-
-If you're new to
-pipelines, see the conceptual guides to [pipelines](/docs/components/pipelines/concepts/pipeline/)
-and [components](/docs/components/pipelines/concepts/component/).
-
-This tutorial describes the manual way of writing a full component program (in any language) and a component definition for it.
-For quickly building component from a python function see [Build component from Python function](/docs/components/pipelines/sdk/lightweight-python-components/) and [Data Passing in Python components](https://github.com/kubeflow/pipelines/blob/master/samples/tutorials/Data%20passing%20in%20python%20components.ipynb).
-
-## Summary
-
-Below is a summary of the steps involved in creating and using a component:
-
-1.  Write the program that contains your component's logic. The program must
-    use files and command-line arguments to pass data to and from the component.
-1.  Containerize the program.
-1.  Write a component specification in YAML format that describes the
-    component for the Kubeflow Pipelines system.
-1.  Use the Kubeflow Pipelines SDK to load your component, use it in a pipeline and run that pipeline.
-
-The rest of this page gives some explanation about input and output data,
-followed by detailed descriptions of the above steps.
-
-## Components and data passing
-
-The concept of component is very similar to the concept of a function.
-Every component can have inputs and outputs (which must be declared in the component specification).
-The component code takes the data passed to its inputs and produces the data for its outputs.
-A pipeline then connects component instances together by passing data from outputs of some components to inputs of other components.
-That's very similar to how a function calls other functions and passes the results between them.
-The Pipelines system handles the actual data passing while the components are responsible for consuming the input data and producing the output data.
-
-## Passing the data to and from the containerized program
 
-When creating a component you need to think about how the component
-will communicate with upstream and downstream components. That is, how it will consume
-the input data and produce the output data.
+A pipeline component is a self-contained set of code that performs one step in
+your ML workflow. This document describes the concepts required to build
+components, and demonstrates how to get started building components.
 
-### Producing data
+## Before you begin
 
-To output any piece of data, the component program must write the output data to some location and inform the system
```

Comment:
In talking with customers and users, this is one of the most important part to understand. People have hard time understanding how to write a component until this "clicks".
Data passing is the most important part to understand about components (since component specifications are basically about describing it).
I'm not sure we should remove this and the next sections.

Reply:
The guidance in these sections is included in the doc. I've revised the **Designing a pipeline component** section to ensure that all the guidance is there.